ios/GoodDollar/AppDelegate.m

```diff
@@ -13,12 +13,15 @@
 #import <TSBackgroundFetch/TSBackgroundFetch.h>
 #import <UserNotifications/UserNotifications.h>
 #import <RNBranch/RNBranch.h>
+#import <Firebase.h>
 
 @implementation AppDelegate
 
 - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
 {
-  
+  if ([FIRApp defaultApp] == nil) {
+    [FIRApp configure];
+  }
```

Comment:
In general AppDelegate's `application` methods looks very overloaded with the different initializations.
It would be good to split the init logic between the different methods. At least - to separate the RN app initialization from the plugins.

I could propose the following solution (just a draft, need to test it and fix possible compilation /runtime errors):
```objectivec
/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

// Delegate definition
#import "AppDelegate.h"

// React imports
#import <React/RCTBridge.h>
#import <React/RCTBundleURLProvider.h>
#import <React/RCTRootView.h>

// React native plugins / services
#import <TSBackgroundFetch/TSBackgroundFetch.h>
#import <UserNotifications/UserNotifications.h>
#import <RNBranch/RNBranch.h>
#import <Firebase.h>

@implementation AppDelegate

- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
{
  [self initAnalytics];
  [self initBranch];
  
  RCTBridge *bridge = [[RCTBridge alloc] initWithDelegate:self launchOptions:launchOptions];
  RCTRootView *rootViewController = [self initRootViewController:bridge];

  self.window = [self initWindow:rootViewController];
  
  [self initBackgroundFetch];
  [self initNotifications];
  
  return YES;
}

- (NSURL *)sourceURLForBridge:(RCTBridge *)bridge
{
#if DEBUG
  return [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index" fallbackResource:nil];
#else
  return [[NSBundle mainBundle] URLForResource:@"main" withExtension:@"jsbundle"];
#endif
}

- (RCTRootView *) initRootViewController:(RCTBridge *)bridge
{
  RCTRootView *rootView = [[RCTRootView alloc] initWithBridge:bridge moduleName:@"GoodDollar" initialProperties:nil];
  UIViewController *rootViewController = [UIViewController new];

  rootView.backgroundColor = [[UIColor alloc] initWithRed:1.0f green:1.0f blue:1.0f alpha:1];
  rootViewController.view = rootView;
  
  return rootView;
}

- (UIWindow *) initWindow:(UIViewController *)rootViewController
{
  (UIWindow *) window = [[UIWindow alloc] initWithFrame:[UIScreen mainScreen].bounds];
  
  window.rootViewController = rootViewController;
  [window makeKeyAndVisible];
}

- (void) initAnalytics
{
  if ([FIRApp defaultApp] == nil) {
    [FIRApp configure];
  }
}

- (void) initBranch
{
  [RNBranch initSessionWithLaunchOptions:launchOptions isReferrable:YES];
}

- (void) initBackgroundFetch
{
  [[TSBackgroundFetch sharedInstance] didFinishLaunching];
}

- (void) initNotifications
{
  UNUserNotificationCenter *center = [UNUserNotificationCenter currentNotificationCenter];
  
  center.delegate = self;
}

@end
```

Reply:
also you will need to update AppDelegate.h:

```objectivec
/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
#import <UserNotifications/UNUserNotificationCenter.h>
#import <React/RCTBridgeDelegate.h>
#import <UIKit/UIKit.h>

@interface AppDelegate : UIResponder <UIApplicationDelegate, RCTBridgeDelegate, UNUserNotificationCenterDelegate>


@property (nonatomic, strong) UIWindow *window;

- (RCTRootView *) initRootViewController:(RCTBridge *)bridge;
- (UIWindow *) initWindow:(UIViewController *)rootViewController;
- (void) initAnalytics;
- (void) initBranch;
- (void) initBackgroundFetch;
- (void) initNotifications;

@end
```